channeldb/invoices.go

```diff
@@ -112,6 +127,33 @@ type Invoice struct {
 	Terms ContractTerm
 }
 
+// getPaymentHash retrieves the payment hash for a given invoice,
+// either by calculating it from the preimage, or using the given
+// hash for invoices with external preimages.
+func getPaymentHash(i *Invoice) ([32]byte, error) {
+	var paymentHash [32]byte
+
+	if i.Terms.ExternalPreimage {
+		if bytes.Equal(i.Terms.PaymentHash[:], zeroHash[:]) {
+			return zeroHash, fmt.Errorf("Invoices with ExternalPreimage must " +
+				"have a locally defined PaymentHash.")
+		}
+
+		// For external preimages, we rely on a provided hash
+		paymentHash = i.Terms.PaymentHash
+	} else {
+		if bytes.Equal(i.Terms.PaymentPreimage[:], zeroPreimage[:]) {
+			return zeroHash, fmt.Errorf("Invoices must have a preimage or" +
+				"use ExternalPreimages")
```

Comment:
Change this error message to say "Invoices must have a preimage or `ExternalPreimages` must be set to true"